code/modules/mob/living/simple_animal/abnormality/waw/contract.dm

```diff
@@ -51,9 +84,48 @@
 	spawned.color = "#000000"	//Make it black to look cool
 	spawned.name = "???"
 	spawned.desc = "What is that thing?"
+	spawned.faction = list("hostile")
 	datum_reference.qliphoth_change(2)
 
 /* Work effects */
+/mob/living/simple_animal/hostile/abnormality/contract/SuccessEffect(mob/living/carbon/human/user, work_type, pe)
+	if(!(user in total_havers))
+		switch(work_type)
+			if(ABNORMALITY_WORK_INSTINCT)
+				if(fort_havers.len < total_per_contract)
+					user.adjust_attribute_buff(FORTITUDE_ATTRIBUTE, (fort_havers.len - 3)*-1 )
+					fort_havers |= user
+					total_havers |= user
+					say("Just sign here on the dotted line... and I'll take care of the rest.")
+
+			if(ABNORMALITY_WORK_INSIGHT)
+				if(prud_havers.len < total_per_contract)
+					user.adjust_attribute_buff(PRUDENCE_ATTRIBUTE, (prud_havers.len - 3)*-1 )
+					prud_havers |= user
+					total_havers |= user
+					say("Just sign here on the dotted line... and I'll take care of the rest.")
+
+			if(ABNORMALITY_WORK_ATTACHMENT)
+				if(temp_havers.len < total_per_contract)
+					user.adjust_attribute_buff(TEMPERANCE_ATTRIBUTE, (temp_havers.len - 3)*-1 )
+					temp_havers |= user
+					total_havers |= user
+					say("Just sign here on the dotted line... and I'll take care of the rest.")
+
+			if(ABNORMALITY_WORK_REPRESSION)
+				if(just_havers.len < total_per_contract)
+					user.adjust_attribute_buff(JUSTICE_ATTRIBUTE, (just_havers.len - 3)*-1 )
+					just_havers |= user
+					total_havers |= user
+					say("Just sign here on the dotted line... and I'll take care of the rest.")
+
+	return
```

Comment:
```suggestion
	if((user in total_havers))
		return
	switch(work_type)
		if(ABNORMALITY_WORK_INSTINCT)
			if(fort_havers.len < total_per_contract)
				user.adjust_attribute_buff(FORTITUDE_ATTRIBUTE, (fort_havers.len - 3)*-1 )
				fort_havers |= user
			else
				return

		if(ABNORMALITY_WORK_INSIGHT)
			if(prud_havers.len < total_per_contract)
				user.adjust_attribute_buff(PRUDENCE_ATTRIBUTE, (prud_havers.len - 3)*-1 )
				prud_havers |= user
			else
				return

		if(ABNORMALITY_WORK_ATTACHMENT)
			if(temp_havers.len < total_per_contract)
				user.adjust_attribute_buff(TEMPERANCE_ATTRIBUTE, (temp_havers.len - 3)*-1 )
				temp_havers |= user
			else
				return
				
		if(ABNORMALITY_WORK_REPRESSION)
			if(just_havers.len < total_per_contract)
				user.adjust_attribute_buff(JUSTICE_ATTRIBUTE, (just_havers.len - 3)*-1 )
				just_havers |= user
			else
				return
				
	total_havers |= user
	say("Just sign here on the dotted line... and I'll take care of the rest.")
	return
```
This should reduce redundancy and also uses early returns to cut off unnecessary code and ideally improve readability.